gridcapa-core-valid-app/src/main/java/com/farao_community/farao/gridcapa_core_valid/app/services/MinioAdapter.java

```diff
@@ -77,15 +76,41 @@ public String generatePreSignedUrl(String filePath) {
         }
     }
 
-    public CoreValidFileResource generateFileResource(String filePath) {
-        try {
-            String fullFilePath = String.format(FORMAT_URL, basePath, filePath);
-            String filename = FilenameUtils.getName(filePath);
-            LOGGER.info("Generates pre-signed URL for file '{}' in Minio bucket '{}'", fullFilePath, bucket);
-            String url = client.getPresignedObjectUrl(GetPresignedObjectUrlArgs.builder().bucket(bucket).object(fullFilePath).expiry(DEFAULT_DOWNLOAD_LINK_EXPIRY_IN_DAYS, TimeUnit.DAYS).method(Method.GET).build());
-            return new CoreValidFileResource(filename, url);
-        } catch (Exception e) {
-            throw new CoreValidInternalException("Exception in MinIO connection.", e);
+    public void deleteCgmBeforeRao(String prefix) {
+        Iterable<Result<Item>> files = listArtifacts(prefix);
+
+        for (Result<Item> result : files) {
+            try {
+                String objectName = result.get().objectName();
+                client.removeObject(RemoveObjectArgs.builder().bucket(bucket).object(objectName).build());
+                LOGGER.info("File {} deleted from Minio", objectName);
+            } catch (Exception e) {
+                LOGGER.error(String.format("Can not delete artifact starting with %s.", prefix));
+            }
+        }
+    }
+
+    private Iterable<Result<Item>> listArtifacts(String prefix) {
```

Comment:
in fact I would make this method public, and I would move the two other ones to CoreValidHandler. Indeed MinioAdapter will soon become a shared dependency with other projects so it is better not to put Core-valid specific code in it.